aspnetcore/fundamentals/host/hosted-services/samples/3.x/BackgroundTasksSample/Services/BackgroundTaskQueue.cs

```diff
@@ -1,42 +1,38 @@
 using System;
-using System.Collections.Concurrent;
 using System.Threading;
+using System.Threading.Channels;
 using System.Threading.Tasks;
 
 namespace BackgroundTasksSample.Services
 {
     #region snippet1
     public interface IBackgroundTaskQueue
     {
-        void QueueBackgroundWorkItem(Func<CancellationToken, Task> workItem);
+        ValueTask QueueBackgroundWorkItemAsync(Func<CancellationToken, ValueTask> workItem);
 
-        Task<Func<CancellationToken, Task>> DequeueAsync(
+        ValueTask<Func<CancellationToken, ValueTask>> DequeueAsync(
             CancellationToken cancellationToken);
     }
 
     public class BackgroundTaskQueue : IBackgroundTaskQueue
```

Comment:
I see we never complete the channel, that works fine here because we're cancelling the reads so it's not a big deal but it made me wonder if we should nuke this background task abstraction altogether... Maybe in another PR

Reply:
👍🏻